wordpress-mobile/WordPress-FluxC-Android
# [REST API] Improve site discovery to work with multiple Link headers

This PR adds support for parsing the REST endpoint even when the site returns multiple `Link` headers when replying to the HEAD request, this uses the native support in `Volley` by reading the property `allHeaders` of the response.

### Testing
1. To prepare the website, please install the [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin, then add the following snippet:
```PHP
function modify_pre_get_shortlink_defaults($return, $id, $context, $allow_slugs) { 
	if ($id == 0) {
		return home_url( '/' );
	}
    return $return; 
}
// Add the filter to return the `shortlink` type as the `Link` header for the home page as well
add_filter( "pre_get_shortlink", "modify_pre_get_shortlink_defaults", 10, 4 );
```
2. Open the example app.
3. Start the sign-in process
4. Enter your site credentials, then your site address.
5. Keep the checkbox at the bottom of the dialog unchecked.
6. Test the login
7. Confirm it works as expected.
